core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala

```diff
@@ -408,4 +625,30 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with B
     assert(thirdTaskDescs.size === 0)
     assert(taskScheduler.getExecutorsAliveOnHost("host1") === Some(Set("executor1", "executor3")))
   }
+
+  test("don't update blacklist for shuffle-fetch failures, preemption, denied commits, " +
```

Comment:
Should this test be in TaskSetManagerSuite.scala? It seems like it's really testing the TSM (which is where updateBlacklistForFailedTask gets called) and not the TaskScheduler.